cloudify/workflows/tasks_graph.py

```diff
@@ -68,7 +68,11 @@ def restore(cls, workflow_context, retrieved_graph):
             if op.containing_subgraph:
                 subgraph_id = op.containing_subgraph
                 op.containing_subgraph = None
-                subgraph = ops[subgraph_id]
+                # subgraph can be missing if the operation was failed but
+                # the subgraph was suceeded (which happens only if an
+                # operation runs twice at the same time, which shouldn't
+                # happen, but let's not break in that case)
+                subgraph = ops.get(subgraph_id, graph)
```

Comment:
added this

this only really happens if we try to cancel+resume an operation that didn't get properly cancelled. Then we end up with 2 instances of the same operation running at the same time (which is pretty much an error condition and perhaps we should add a validation that this doesnt happen), and when the first operation finishes and marks the subgraph as done, and we try to resume the 2nd instance of the operation, the subgraph is already done. So, we'll just attach that operation to the main graph, in place of the subgraph.

Reply:
What do u think about adding some debug msg about this for future problems?